modules/framework/cats/test/src-jvm/MetaJVM.scala

```diff
@@ -64,14 +69,24 @@ object MetaJVM {
     def sharedResources(global: weaver.GlobalWrite): Resource[IO, Unit] =
       Resource.eval {
         for {
-          initialised <- Ref[IO].of(0)
-          finalised   <- Ref[IO].of(0)
-          totalUses   <- Ref[IO].of(0)
+          initialised   <- Ref[IO].of(0)
+          finalised     <- Ref[IO].of(0)
+          totalUses     <- Ref[IO].of(0)
+          parallelLatch <- CyclicBarrier[IO](3)
```

Comment:
The thinking here is that the suites could use this latch to make sure that all of them are running at the same time

Once the barrier is reached, they can run the assertions on the state.

The lazy resource will stay open and be re-used, because no tests will finish.

Using a fixed number here is problematic because it hardcodes the physical number of tests running in parallel - if we add another test to `LazyAccessParallel` spec, it will break again.

Reply:
> Using a fixed number here is problematic because it hardcodes the physical number of tests running in parallel 

This is fine, let's just put a comment above it.

Reply:
done